ui: Statement diagnostics history - empty state

vladlos opened the pull request:
Use `SortableTable` component instead of antd table, this change gives us oportunity to reuse realisation of empty and loading states of table on Statement diagnostics history page.
Minor updates to <Empty/> component props api, to avoid duplicates of window.open calls
Added storybook for SortableTable empty state.

Resolves: #46576

Release note (ui): new design of empty state of Statement diagnostics history page

![Screenshot 2020-04-24 at 16 11 20](https://user-images.githubusercontent.com/12850886/80216182-57f92580-8646-11ea-832e-8584b3c6c69f.png)

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/46921)
<!-- Reviewable:end -->

dhartunian reviewed (changes requested):
While I agree that re-using the components is generally the right thing to do, this change modifies the functionality of this page in a big way: previously the links to statement diagnostics would take the you to the statements page.

Additionally the tooltip on a "Ready" diagnostic is now confusing because it tells you to go to the details page but you can't...

Can you make the change without modifying the table itself? Let me know if that's tricky and we can discuss alternatives.

> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46921#-:-M5hPQZ2CNQYMUkRC9xd:bk1pq0)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @elkmaster)

___
*[pkg/ui/src/views/reports/containers/statementDiagnosticsHistory/index.tsx, line 179 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/46921#-M5gxzT45VtVWuH6YwKo:-M5gxzT5Bn2Vk3uBmIxw:bbrss3w) ([raw file](https://github.com/cockroachdb/cockroach/blob/bbe8b794ec19f0641cdadb2b890c1dd456188b2e/pkg/ui/src/views/reports/containers/statementDiagnosticsHistory/index.tsx#L179)):*
> ```XML
>             description: "Statement diagnostics can help when troubleshooting issues with specific queries. The diagnostic bundle can be activated from individual statement pages and will include EXPLAIN plans, table statistics, and traces.",
>             label: "Learn more",
>             onClick: () => window.open(statementsTable),
> ```

can we change the `onClick` to be a link href instead? do we ever need to support a special click action?


<!-- Sent from Reviewable.io -->

dhartunian reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46921)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dhartunian and @elkmaster)

___
*[pkg/ui/src/views/reports/containers/statementDiagnosticsHistory/index.tsx, line 179 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/46921#-M5gxzT45VtVWuH6YwKo:-M5wS6Nz8RZ8ijNkIMK9:bbjf4sg) ([raw file](https://github.com/cockroachdb/cockroach/blob/bbe8b794ec19f0641cdadb2b890c1dd456188b2e/pkg/ui/src/views/reports/containers/statementDiagnosticsHistory/index.tsx#L179)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

can we change the `onClick` to be a link href instead? do we ever need to support a special click action?
</blockquote></details>

I don't think there's a need to have there be an `onClick` function option at all.
Can we change the name to `buttonHref` or something and make it always a string? Then in the `Empty` component you can wrap the `Button` in an `Anchor`. Is that possible?
___
*[pkg/ui/src/views/reports/containers/statementDiagnosticsHistory/index.tsx, line 85 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/46921#-M5wVgWa0Y1XmY0dmuLC:-M5wVgWa0Y1XmY0dmuLD:bim7ov) ([raw file](https://github.com/cockroachdb/cockroach/blob/09afd2f1d4d4c8fcb39eb8434562a1c4ccf839bb/pkg/ui/src/views/reports/containers/statementDiagnosticsHistory/index.tsx#L85)):*
> ```XML
> 
> class StatementDiagnosticsHistoryView extends React.Component<StatementDiagnosticsHistoryViewProps, StatementDiagnosticsHistoryViewState> {
>   columns: ColumnDescriptor<IStatementDiagnosticsReport>[] = [
> ```

Can we set the mouse icon to "pointer" on the column headers?
___
*[pkg/ui/src/views/reports/containers/statementDiagnosticsHistory/index.tsx, line 116 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/46921#-M5wVdaS2eybzwJTX_-J:-M5wVdaS2eybzwJTX_-K:b-vuymzu) ([raw file](https://github.com/cockroachdb/cockroach/blob/09afd2f1d4d4c8fcb39eb8434562a1c4ccf839bb/pkg/ui/src/views/reports/containers/statementDiagnosticsHistory/index.tsx#L116)):*
> ```XML
>     {
>       title: "Status",
>       sort: record => record.completed.toString(),
> ```

Clicking on status sort breaks the page
___
*[pkg/ui/src/views/reports/containers/statementDiagnosticsHistory/index.tsx, line 164 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/46921#-M5wVnyM9BxBR8xlepfP:-M5wVnyM9BxBR8xlepfQ:b4fmnv3) ([raw file](https://github.com/cockroachdb/cockroach/blob/09afd2f1d4d4c8fcb39eb8434562a1c4ccf839bb/pkg/ui/src/views/reports/containers/statementDiagnosticsHistory/index.tsx#L164)):*
> ```XML
>       sortSetting: {
>         sortKey: 0,
>         ascending: true,
> ```

The default sort order is descending. we'd like to see the newest diagnostics at the top by default.
___
*[pkg/ui/src/views/reports/containers/statementDiagnosticsHistory/index.tsx, line 230 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/46921#-M5wSn5l33L7uDLxHMws:-M5wSn5l33L7uDLxHMwt:bfrs83s) ([raw file](https://github.com/cockroachdb/cockroach/blob/09afd2f1d4d4c8fcb39eb8434562a1c4ccf839bb/pkg/ui/src/views/reports/containers/statementDiagnosticsHistory/index.tsx#L230)):*
> ```XML
>             description: "Statement diagnostics can help when troubleshooting issues with specific queries. The diagnostic bundle can be activated from individual statement pages and will include EXPLAIN plans, table statistics, and traces.",
>             label: "Learn more",
>             onClick: statementsTable,
> ```

This link should be to `statementDiagnostics`, not `statementsTable`.


<!-- Sent from Reviewable.io -->

elkmaster commented:
> Clicking on status sort breaks the page
couldn't recreate this on, my dataset. provided fix by guess, should be retested.

onClick function option is needed, but also added `buttonHref` to avoid inconvenience in naming. Added ts rule to accept only one of that props at a time.

dhartunian reviewed (changes requested):
@elkmaster I expected to see the empty state at https://localhost:3000/#/reports/statements/diagnosticshistory when running this. Instead I still see:

![Screen Shot 2020-05-26 at 6 42 44 PM](https://user-images.githubusercontent.com/986307/82957316-b33d7100-9f80-11ea-8680-23a1cc1fc599.png)

Let me know if I'm missing something.
Please add a Storybook story rendering the empty table with the empty state showing.

Additionally, please describe in the commit message, PR, and Release Notes which pages will have modified behavior and in which cases we expect the empty state to show. Clarify that the empty state provides documentation to customers.

elkmaster commented:
@dhartunian, sorry just a typo. Should be fixed now. Updated description and added storybook for empty table. In this visible updates will be only on `Statement diagnostics history page`.

dhartunian reviewed (approved):
@elkmaster LGTM. Thanks for making the latest round of changes. Please rebase and resolve conflicts, then merge with bors.

vladlos committed:
ui: Statement diagnostics history - empty state

Use `SortableTable` component instead of antd table, this change gives us oportunity to reuse realisation of empty and loading states of table on Statement diagnostics history page.
Minor updates to <Empty/> component props api, to avoid duplicates of window.open calls
Added storybook for SortableTable empty state.

Resolves: #46576

Release note (ui): new design of empty state of Statement diagnostics history page

elkmaster commented:
bors r+
